fvm/evm/stdlib/contract.go

```diff
@@ -38,6 +40,272 @@ var evmAddressBytesType = sema.NewConstantSizedType(nil, sema.UInt8Type, types.A
 var evmAddressBytesStaticType = interpreter.ConvertSemaArrayTypeToStaticArrayType(nil, evmAddressBytesType)
 var EVMAddressBytesCadenceType = cadence.NewConstantSizedArrayType(types.AddressLength, cadence.TheUInt8Type)
 
+// EVM.encodeABI
+
+const internalEVMTypeEncodeABIFunctionName = "encodeABI"
+
+var internalEVMTypeEncodeABIFunctionType = &sema.FunctionType{
+	Parameters: []sema.Parameter{
+		{
+			Label:      sema.ArgumentLabelNotRequired,
+			Identifier: "values",
+			TypeAnnotation: sema.NewTypeAnnotation(
+				sema.NewVariableSizedType(nil, sema.AnyStructType),
+			),
+		},
+	},
+	ReturnTypeAnnotation: sema.NewTypeAnnotation(sema.ByteArrayType),
+}
+
+func newInternalEVMTypeEncodeABIFunction(
+	gauge common.MemoryGauge,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewHostFunctionValue(
+		gauge,
+		internalEVMTypeEncodeABIFunctionType,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			inter := invocation.Interpreter
+			locationRange := invocation.LocationRange
+
+			// Get `values` argument
+
+			valuesArray, ok := invocation.Arguments[0].(*interpreter.ArrayValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			values := make([]interface{}, 0)
+			var arguments gethABI.Arguments
+
+			for i := 0; i < valuesArray.Count(); i++ {
+				element := valuesArray.Get(inter, locationRange, i)
+				switch value := element.(type) {
+				case *interpreter.StringValue:
+					values = append(values, value.Str)
+					typ, err := gethABI.NewType("string", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.UInt64Value:
+					values = append(values, uint64(value))
+					typ, err := gethABI.NewType("uint64", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.BoolValue:
+					values = append(values, bool(value))
+					typ, err := gethABI.NewType("bool", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case *interpreter.CompositeValue:
+					if value.QualifiedIdentifier == "EVM.EVMAddress" {
+						bytes, err := interpreter.ByteArrayValueToByteSlice(
+							inter,
+							value.GetMember(inter, locationRange, "bytes"),
+							locationRange,
+						)
+						if err != nil {
+							panic(err)
+						}
+						values = append(values, gethCommon.Address(bytes))
+						typ, err := gethABI.NewType("address", "", nil)
+						if err != nil {
+							panic(err)
+						}
+						arguments = append(arguments, gethABI.Argument{Type: typ})
+					}
+				case *interpreter.ArrayValue:
+					elements := make([]uint64, 0)
```

Comment:
We should also decide on which primitives we are going to support for encoding/decoding arrays. Here I have just added support for `[UInt64]`